### PR TITLE
Added Rabin_karp.js (Fixes issue #2161)

### PR DESCRIPTION
**Pull request by JanaSabuj**
Fixes #2161 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Rabin Karp algorithm implementation in JS.

#### Languages Used:
- JavaScript

#### Files Added:
- Rabin_karp.js

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by JanaSabuj** [comment]
@jainaman224 @asha15 @SakshiShreya please do review the PR closing the Rabin_karp JS issue #2161 .

**Comment by JanaSabuj** [comment]
@jainaman224 @asha15 @SakshiShreya please do review the PR.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 .DS_Store
-.vscode/*
\ No newline at end of file
+.vscode/*
```
Comment (asha15) [inline .gitignore:2]: can you revert this changes?

**Comment by JanaSabuj** [comment]
@asha15 please do review the PR

File: .gitignore (hunk 1 above)
Comment (asha15) [reply .gitignore:2]: ?

**Comment by JanaSabuj** [comment]
@asha15 done. please review.

**Comment by asha15** [comment]
please create new PR asap with the relevant file

**Comment by JanaSabuj** [comment]
@asha15 I created a new PR.
